src/main/java/stirling/software/SPDF/config/security/InitialSecuritySetup.java

```diff
@@ -1,49 +1,48 @@
 package stirling.software.SPDF.config.security;
 
-import java.io.IOException;
+import java.sql.SQLException;
 import java.util.UUID;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import jakarta.annotation.PostConstruct;
 import lombok.extern.slf4j.Slf4j;
-import stirling.software.SPDF.config.interfaces.DatabaseBackupInterface;
+import stirling.software.SPDF.config.interfaces.DatabaseInterface;
 import stirling.software.SPDF.model.ApplicationProperties;
 import stirling.software.SPDF.model.Role;
+import stirling.software.SPDF.model.provider.UnsupportedProviderException;
 
-@Component
 @Slf4j
+@Component
 public class InitialSecuritySetup {
 
-    private final UserService userService;
-
-    private final ApplicationProperties applicationProperties;
+    @Autowired private UserService userService;
 
-    private final DatabaseBackupInterface databaseBackupHelper;
+    @Autowired private ApplicationProperties applicationProperties;
 
-    public InitialSecuritySetup(
-            UserService userService,
-            ApplicationProperties applicationProperties,
-            DatabaseBackupInterface databaseBackupHelper) {
-        this.userService = userService;
-        this.applicationProperties = applicationProperties;
-        this.databaseBackupHelper = databaseBackupHelper;
-    }
+    @Autowired private DatabaseInterface databaseService;
```

Comment:
Switch to using the autowired via constructor instead (I think you removed this in a merge?)